### PR TITLE
core(fr): align base artifacts with legacy gather-runner

### PR DESCRIPTION
**Pull request by patrickhulce**
**Summary**
Brings the base artifacts support in FR up to snuff with legacy gather runner.

**Related Issues/PRs**
ref #11313

### CODE REVIEW

File: lighthouse-core/test/fraggle-rock/api-test-pptr.js [hunk 1]
```diff
@@ -158,8 +158,8 @@ describe('Fraggle Rock API', () => {
 
       const {lhr} = result;
       const {auditResults, failedAudits, erroredAudits} = getAuditsBreakdown(lhr);
-      // TODO(FR-COMPAT): This assertion can be removed when full compatibility is reached.
-      expect(auditResults.length).toMatchInlineSnapshot(`112`);
+      // TODO(FR-COMPAT): This assertion can be removed when full compatibility is reached. HEAD
+      expect(auditResults.length).toMatchInlineSnapshot(`141`);
```
Comment (adamraine) [inline lighthouse-core/test/fraggle-rock/api-test-pptr.js:162]: 👀

**Review by adamraine — COMMENTED** [review]
Will base artifacts be available in gatherers as dependencies, or can we just pass them in `FRTransitionalContext` separately?

**Comment by patrickhulce** [comment]
> Will base artifacts be available in gatherers as dependencies, or can we just pass them in FRTransitionalContext separately?

Great question, this does not yet enable base artifacts as dependencies. We can talk about how to accomplish that when the need arises (has it already come up for you?). URL is the biggest one I can think of, but that's already available on the phaseContext.

**Comment by adamraine** [comment]
> We can talk about how to accomplish that when the need arises (has it already come up for you?)

Yes, I know `script-elements` uses `HostFormFactor`.

I think it would be easier to include base artifacts separately in the context, or include them as dependencies for all gatherers by default. Essentially I want to avoid writing:
```js
/** @type {LH.Gatherer.FRTransitionalContext<'HostFormFactor'>} */
meta = {
	supportedModes = ['navigation'],
	dependencies: {HostFormFactor: HostFormFactor.symbol},
}
```

**Review by adamraine — APPROVED** [review]
For our current set of Gatherers, it looks like we only need to depend on `HostFormFactor` and `Settings`. Explicit dependencies probably make sense here since there are so few instances of us depending on base artifacts.

**Pull request merged**